artemis-journal/src/main/java/org/apache/activemq/artemis/core/io/nio/NIOSequentialFile.java

```diff
@@ -173,6 +173,40 @@ public int read(final ByteBuffer bytes) throws Exception {
       return read(bytes, null);
    }
 
+   //On *nix platform is the size by which RandomAccessFile write/read are performed using a
+   //stack-allocated copy of the java byte[]
```

Comment:
And what about windows

Reply:
This was just a POC commit to mitigate the issue , but we can discuss on the comments about it :)
Thanks to looking at it

Reply:
I've verified that is a the same behaviour for Windows: it is in the shared JNI code by any OSs